automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/PublishedWorkspaceSpec.scala

```diff
@@ -0,0 +1,184 @@
+            withWorkspace(billingProject, "PublishedWorkspaceSpec_workspace") { workspaceName =>
+
+              val data = LibraryData.metadataBasic + ("library:datasetName" -> workspaceName)
+              Orchestration.library.setLibraryAttributes(billingProject, workspaceName, data)
+              Orchestration.library.publishWorkspace(billingProject, workspaceName)
+
+              val clonedWorkspaceName = uuidWithPrefix("cloneWorkspace")
+
+              register cleanUp Orchestration.workspaces.delete(billingProject, clonedWorkspaceName)
+              Orchestration.workspaces.clone(billingProject, workspaceName, billingProject, clonedWorkspaceName)
+              // check cloned workspace does not have published status in workspace detail: "library:published": false
+              val response = Rawls.workspaces.getWorkspaceDetails(billingProject, clonedWorkspaceName)
+              response should include(""""library:published":false""")
+
+              // check cloned workspace is not visible in library
+              withClue("a cloned workspace should not be visible in library.") {
+                eventually {
+                  isVisibleInLibrary(clonedWorkspaceName) shouldBe false
+                }
+              }
+            }
+          }
+        }
+
+        "should default to visible to 'all users' in Library Attributes" in {
+
+          val curatorUser = UserPool.chooseCurator
+          implicit val curatorAuthToken: AuthToken = curatorUser.makeAuthToken()
+
+          withCleanBillingProject(curatorUser) { billingProject =>
+            withWorkspace(billingProject, "PublishedWorkspaceSpec_workspace") { workspaceName =>
+
+              val data = LibraryData.metadataBasic + ("library:datasetName" -> workspaceName)
+              Orchestration.library.setLibraryAttributes(billingProject, workspaceName, data)
+              Orchestration.library.setDiscoverableGroups(billingProject, workspaceName, List("all_broad_users"))
+              Orchestration.library.publishWorkspace(billingProject, workspaceName)
+
+              val clonedWorkspaceName = workspaceName + "_clone"
+              register cleanUp Orchestration.workspaces.delete(billingProject, clonedWorkspaceName)
+              Orchestration.workspaces.clone(billingProject, workspaceName, billingProject, clonedWorkspaceName)
+
+              //Verify default group "All users"
+              //In swagger you make sure that getDiscoverableGroup endpoint shows []
+              withClue(s"Get api/library/${billingProject}/${clonedWorkspaceName}/discoverableGroups endpoint shows []") {
+                eventually {
+                  val accessGroup: Seq[String] = Orchestration.library.getDiscoverableGroups(billingProject, clonedWorkspaceName)
+                  accessGroup.size shouldBe 0
+                }
+              }
+            }
+          }
+        }
+
+      }
+    }
+
+    "For a dataset with consent codes" in {
```

Comment:
This test name and the name of the next test should be more descriptive of what the tests are doing. Currently, the names of the tests are `For a user with publish permissions For a dataset with consent codes` and `For a user with publish permissions For a dataset with tags` which don't tell me much about the test. I know these are the names of the tests already, but it'd be great if we could rename them to `A user with publish permissions can publish a dataset with consent codes` or something along those lines

Reply:
👍 updated names. let me know if new names still confusing.